zipline/pipeline/factors/technical.py

```diff
@@ -683,3 +679,115 @@ def compute(self, today, assets, out, highs, lows, closes):
+        The window length for the "fast" EWMA. Default is 12.
+    slow_period : int >= 0, <= window_length
+        The window length for the "slow" EWMA. Default is 26.
+    signal_period' : int >= 0, <= slow_period
+        The window length for the signal line. Default is 9.
+
+    Returns
+    -------
+    MACD: The difference between "fast" EWMA and "slow" EWMA.
+    signal: The EWMA of the MACD line using `signal_period` as span.
+    hist: Difference between MACD and signal. (Divergence series)
+    """
+    inputs = [USEquityPricing.close]
+    params = ('fast_period', 'slow_period', 'signal_period')
+    outputs = ('MACD', 'signal', 'hist')
+
+    def __new__(cls,
+                fast_period=12,
+                slow_period=26,
+                signal_period=9,
+                *args,
+                **kwargs):
+        return super(MovingAverageConvergenceDivergence, cls).__new__(
+            cls,
+            fast_period=fast_period,
+            slow_period=slow_period,
+            signal_period=signal_period,
+            window_length=slow_period + signal_period - 1,
+            *args, **kwargs
+        )
+
+    def calculate_ewma(self, data, length):
+        decay_rate = 1.0 - (2.0 / (1.0 + length))
+        return average(data,
+                       axis=1,
+                       weights=self.weights(length, decay_rate))
+
+    def calculate_macd(self, col):
+        slow_EWMA = self.calculate_ewma(
+            rolling_window(
+                col,
+                self.params['slow_period']
+            ),
+            self.params['slow_period'])
+        fast_EWMA = self.calculate_ewma(
+            rolling_window(
+                col,
+                self.params['fast_period']
+            )[-self.params['signal_period']:],
+            self.params['fast_period'])
+        macd = fast_EWMA - slow_EWMA
+        signal_line = self.calculate_ewma(
+            macd.reshape(-1, self.params['signal_period']),
+            self.params['signal_period'])
+        hist = macd[-1] - signal_line
+        return macd[-1], signal_line[-1], hist[-1]
+
+    def compute(self, today, assets, out, close, fast_period, slow_period,
+                signal_period):
+        macd, sig, hist = zip(*map(self.calculate_macd, close.T))
+        out.macd[:] = macd
+        out.signal[:] = sig
+        out.hist[:] = hist
+
+
+class AnnualizedVolatility(CustomFactor):
+    """
+    Volatility
+    https://en.wikipedia.org/wiki/Volatility_(finance)
+
+    The degree of variation of a series over time as measured by the standard
+    deviation of returns.
+
+    **Default Inputs:**
+        :data:`zipline.pipeline.factors.Returns(window_length=2)`
+
+    Parameters
+    ----------
+    annualization_factor :
+        The number of time units per year. Defaults to average number of NYSE
+        trading days per year, 252.
+    """
+    inputs = [Returns(window_length=2)]
+    params = {'annualization_factor': 252}
```

Comment:
When would you not want `annualization_factor` to match the window length?  If we want to make the length configurable, more natural to me would just be to have:

```
class Volatility(CustomFactor):
    inputs = [Returns(...)]

    def compute(self, today, assets, out):
        out[:] =  nanstd(returns, ddof=0, axis=0) * (self.window_length ** .5)
```